scipy/linalg/tests/test_matfuncs.py

```diff
@@ -1045,3 +1048,13 @@ def test_empty(self):
         b = np.empty((5, 0))
         res = khatri_rao(a, b)
         assert_allclose(res, np.empty((15, 0)))
+
+@pytest.mark.parametrize('func', [cosm, sinm, tanm, coshm, sinhm, tanhm])
+@pytest.mark.parametrize('dt', [int, float, np.float32, complex, np.complex64])
+def test_empty_matrix_input(func, dt):
```

Comment:
Actually, since all the functions that rely on `expm` aren't tested separately from `expm`, I'll just move this function into the `TestExpm` class and test them all at once to reduce code duplication.